apps/vaporgui/FlowSubtabs.h

```diff
@@ -111,49 +108,96 @@ class FlowSeedingSubtab : public QVaporSubtab {
-    void _rakeBiasStrengthChanged();
+    void _pathlineLengthChanged( int length );
+    void _pathlineStartTimeChanged( int startTime );
+    void _pathlineEndTimeChanged( int endTime );
+    void _pathlineLifetimeChanged( int lifeTime );
 
-    void _steadyGotClicked();
-    void _steadyNumOfStepsChanged();
-    void _pastNumOfTimeStepsChanged( int );
-    void _seedInjIntervalChanged( int );
-    
-    void _selectedTabChanged(int index);
+    void _streamlineDirectionChanged( int index );
+    void _streamlineSamplesChanged( int length );
 
-private:
-    VAPoR::FlowParams*      _params;
-    VAPoR::ParamsMgr *      _paramsMgr;
+    void _seedInjIntervalChanged( int interval );
+
+    void _periodicClicked();
+    void _velocityMultiplierChanged( const std::string& multiplier );
 
-    /* Add some QT widgets */
-    VCheckBox*              _steady;
-    VLineEdit*              _steadyNumOfSteps;
-    VIntSlider*             _pastNumOfTimeSteps;
-    VIntSlider*             _seedInjInterval;
+    void _seedGenModeChanged( int newIdx );
+    void _rakeNumOfSeedsChanged();
+    void _seedListFileChanged( const std::string& file );
+    void _biasVariableChanged( const std::string& variable );
+    void _biasStrengthChanged( double bias );
 
-    VComboBox*              _seedGenMode;
-    VFileReader*            _fileReader;
-    VFileWriter*            _fileWriter;
-    VComboBox*              _flowDirection;
+    void _rakeGeometryChanged( const std::vector<float>& range );
 
-    /* Rake related widgets */
-    VGeometry*              _rake;
-    VLineEdit              *_rakeXNum, *_rakeYNum, *_rakeZNum, *_rakeTotalNum;
-    VComboBox*              _rakeBiasVariable;
-    VSlider*                _rakeBiasStrength;
+    void _geometryWriterClicked();
 
-    QFrame                  *_hline1, *_hline2;     // horizontal lines
+    void _selectedTabChanged( int index );
+
+private:
+    void _createIntegrationSection();
+    void _createSeedingSection( QWidget* parent );
+    void _updateStreamlineWidgets( VAPoR::DataMgr* dataMgr);
+    void _updatePathlineWidgets( VAPoR::DataMgr* dataMgr);
+
+    VAPoR::FlowParams*      _params;
+    VAPoR::ParamsMgr *      _paramsMgr;
 
-    /* Helper functions */
-    void _hideShowWidgets(); // hide and show widgets based on the current seed generation mode.
+    // Integration options
+    VSection*               _integrationSection;
+    VComboBox*              _flowTypeCombo;
+
+    //  Streamline integration options
+    VFrame*                 _streamlineFrame;
+    VSliderEdit*            _streamlineSamplesSliderEdit;
+    VComboBox*              _streamlineDirectionCombo;
+
+    //  Pathline integration options
+    VFrame*                 _pathlineFrame;
+    VSliderEdit*            _pathlineLengthSliderEdit;
+    VSliderEdit*            _pathlineInjIntervalSliderEdit;
+    VSliderEdit*            _pathlineStartSliderEdit;
+    VSliderEdit*            _pathlineEndSliderEdit;
```

Comment:
Can we remove (or at least comment out) GUI elements that ain't supported by the backend yet?

Reply:
this is done